fanficfare/adapters/adapter_deviantartcom.py

```diff
@@ -0,0 +1,187 @@
+        self._setURL(url)
+
+    @staticmethod
+    def getSiteDomain():
+        return 'www.deviantart.com'
+
+    @classmethod
+    def getAcceptDomains(cls):
+        return ['www.deviantart.com']
+
+    @classmethod
+    def getProtocol(self):
+        return 'https'
+
+    @classmethod
+    def getSiteExampleURLs(cls):
+        return 'https://%s/<author>/art/<work-name>' % cls.getSiteDomain()
+
+    def getSiteURLPattern(self):
+        return r'https?://www\.deviantart\.com/(?P<author>[^/]+)/art/(?P<id>[^/]+)/?'
+
+    def performLogin(self, url):
+        data = self.get_request_raw('https://www.deviantart.com/users/login', referer=url)
+        data = self.decode_data(data)
+        soup = self.make_soup(data)
+        params = {
+            'referer': url,
+            'csrf_token': soup.find('input', {'name': 'csrf_token'})['value'],
+            'challenge': soup.find('input', {'name': 'challenge'})['value'],
+        }
+
+        if self.password:
+            params['username'] = self.username
+            params['password'] = self.password
+        else:
+            params['username'] = self.getConfig('username')
+            params['password'] = self.getConfig('password')
+
+        loginUrl = 'https://' + self.getSiteDomain() + '/_sisu/do/signin'
+        logger.debug('Will now login to deviantARt as (%s)' % params['username'])
+
+        result = self.post_request(loginUrl, params, usecache=False)
+
+        if 'Log In | DeviantArt' in result:
+            logger.error('Failed to login to deviantArt as %s' % params['username'])
+            raise exceptions.FailedToLogin('https://www.deviantart.com', params['username'])
+        else:
+            return True
+
+    def requiresLogin(self, data):
+        return '</a> has limited the viewing of this artwork to members of the DeviantArt community only' in data
+
+    def isWatchersOnly(self, data):
+        return '<span>Watchers-Only Deviation</span>' in data
+
+    def requiresMatureContentEnabled(self, data):
+        return (
+            '>This content is intended for mature audiences<' in data
+            or '>This deviation is intended for mature audiences<' in data
+        )
+
+    def extractChapterUrlsAndMetadata(self):
+        isLoggedIn = False
+        logger.debug('URL: %s', self.url)
+
+        data = self.get_request(self.url)
+
+        soup = self.make_soup(data)
+
+        if self.requiresLogin(data):
+            if self.performLogin(self.url):
+                isLoggedIn = True
+                data = self.get_request(self.url, usecache=False)
+                soup = self.make_soup(data)
+
+        if self.isWatchersOnly(data):
+            raise exceptions.FailedToDownload(
+                'Deviation is only available for watchers.' +
+                'You must watch this author before you can download it.'
+            )
+
+        if self.requiresMatureContentEnabled(data):
```

Comment:
While deviantArt has a "click to show mature content" button, it's entirely done in JavaScript. I couldn't find a URL parameter that would force it to show, and the content isn't inserted into the HTML body until after the button is clicked, so the only solution I could think of here was to error out with `FailedToDownload` and tell the user they need to adjust their account settings.